### ISSUE TITLE
Cannot build documentation for pretty_env_logger crate: "multiple rlib candidates for `ansi_term` found"

### ISSUE THREAD

**Issue by zomerfaun** [rust-lang/rust#49002]
I have included the `pretty_env_logger` crate into my project and now my documentation won't build anymore:
```
zomerfaun@falafel ~/C/grot> cargo +nightly doc
 Documenting failure v0.1.1
 Documenting pretty_env_logger v0.2.2
error[E0465]: multiple rlib candidates for `ansi_term` found
  --> /home/zomerfaun/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
note: candidate #1: /home/zomerfaun/Code/grot/target/debug/deps/libansi_term-e2b3ca16da03f7b2.rlib
  --> /home/zomerfaun/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^
note: candidate #2: /home/zomerfaun/Code/grot/target/debug/deps/libansi_term-aa8dbd83a74bc2d7.rlib
  --> /home/zomerfaun/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^

error[E0463]: can't find crate for `ansi_term`
  --> /home/zomerfaun/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to 2 previous errors

You've got a few errors: E0463, E0465
If you want more information on an error, try using "rustc --explain E0463"
error: Could not document `pretty_env_logger`.

Caused by:
  process didn't exit successfully: `rustdoc --crate-name pretty_env_logger /home/zomerfaun/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs -o /home/zomerfaun/Code/grot/target/doc -L dependency=/home/zomerfaun/Code/grot/target/debug/deps --extern ansi_term=/home/zomerfaun/Code/grot/target/debug/deps/libansi_term-e2b3ca16da03f7b2.rlib --extern ansi_term=/home/zomerfaun/Code/grot/target/debug/deps/libansi_term-aa8dbd83a74bc2d7.rlib --extern env_logger=/home/zomerfaun/Code/grot/target/debug/deps/libenv_logger-53e195ae747b91ed.rlib --extern log=/home/zomerfaun/Code/grot/target/debug/deps/liblog-be8fd45b81578a9c.rlib` (exit code: 101)
warning: build failed, waiting for other jobs to finish...
error: build failed
zomerfaun@falafel ~/C/grot> rustc +nightly --explain E0465
error: no extended information for E0465
```
Above I tried this using nightly, but using stable gives the same result.  There is a relevant issue at https://github.com/seanmonstar/pretty-env-logger/issues/14.

The crate itself (and its inclusion into my project) builds and runs fine, only the documentation fails to build, so that is why I assume this is a rustdoc bug.  At least I assume it should resolve dependencies in the same way that rustc does.  If not then it would be very helpful if rustdoc would continue after the error and finish documenting the other crates, so I can view the documentation for my own crate at least :sweat_smile:

If my assumptions are totally wrong and this is not a bug in rustdoc at all, then I am very sorry.  This is my first bug report so I hope I did everything right.  Thank you for making Rust :smile: 

## Meta
```
zomerfaun@falafel ~/C/grot> rustc --version --verbose
rustc 1.24.1 (d3ae9a9e0 2018-02-27)
binary: rustc
commit-hash: d3ae9a9e08edf12de0ed82af57ba2a56c26496ea
commit-date: 2018-02-27
host: x86_64-unknown-linux-gnu
release: 1.24.1
LLVM version: 4.0
zomerfaun@falafel ~/C/grot> rustc +nightly --version --verbose 
rustc 1.26.0-nightly (2789b067d 2018-03-06)
binary: rustc
commit-hash: 2789b067da2ac921b86199bde21dd231ace1da39
commit-date: 2018-03-06
host: x86_64-unknown-linux-gnu
release: 1.26.0-nightly
LLVM version: 6.0
```

**Comment by dbrgn** [comment]
Same problem on Rust 1.25 stable:

```
$ cargo new foo
$ cd foo
$ echo 'pretty_env_logger = "*"' >> Cargo.toml
$ cargo doc -p pretty_env_logger
...
error[E0465]: multiple rmeta candidates for `ansi_term` found
  --> /home/danilo/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
note: candidate #1: /tmp/foo/target/debug/deps/libansi_term-8f29ff0b2086d6a6.rmeta
  --> /home/danilo/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^
note: candidate #2: /tmp/foo/target/debug/deps/libansi_term-fa0b3ae5de1b2f52.rmeta
  --> /home/danilo/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^

error[E0463]: can't find crate for `ansi_term`
  --> /home/danilo/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs:24:1
   |
24 | extern crate ansi_term;
   | ^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to 2 previous errors

error: Could not document `pretty_env_logger`.

Caused by:
  process didn't exit successfully: `rustdoc --crate-name pretty_env_logger /home/danilo/.cargo/registry/src/github.com-1ecc6299db9ec823/pretty_env_logger-0.2.2/src/lib.rs -o /tmp/foo/target/doc -L dependency=/tmp/foo/target/debug/deps --extern ansi_term=/tmp/foo/target/debug/deps/libansi_term-fa0b3ae5de1b2f52.rmeta --extern env_logger=/tmp/foo/target/debug/deps/libenv_logger-bd42acbe717fae2e.rmeta --extern ansi_term=/tmp/foo/target/debug/deps/libansi_term-8f29ff0b2086d6a6.rmeta --extern log=/tmp/foo/target/debug/deps/liblog-c6a51ef82288204a.rmeta` (exit code: 101)
```

**Comment by derekdreery** [comment]
I'm getting this error on windows, so its not a linux issue.

**Comment by derekdreery** [comment]
I suspect this error originates from the fact that there is target info in the dependencies. It looks like rustdoc is including both versions of ansi_term. I will have a look more when I'm on a computer.

**Comment by derekdreery** [comment]
Fixed by rust-lang/cargo#5345.

**Comment by QuietMisdreavus** [comment]
Closing, as this issue was fixed in Cargo and it's been updated in this repo since then.
